src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaConsumerContext.cs

```diff
@@ -15,64 +15,90 @@ public class KafkaConsumerContext<TKey, TValue> :
         ConsumerContext<TKey, TValue>
         where TValue : class
     {
-        readonly IConsumer<TKey, TValue> _consumer;
+        IConsumer<TKey, TValue>[] _consumers;
```

Comment:
Encapsulate this behavior into a class, related array like this are an anti-pattern.